orttraining/orttraining/core/graph/zero_optimizer_graph_builder.cc

```diff
@@ -93,39 +95,94 @@ static Status AddL2NormNcclAllReduce(
   return Status::OK();
 }
 
+static std::vector<ArgDef> AddPartitionsForParameter(
+    Graph& graph,
+    GraphAugmenter::GraphDefs& graph_defs,
+    const std::string& initializer_name,
+    const std::vector<TensorShape>& shapes,
+    std::unordered_map<std::string, std::string>& updated_weight_names_map){
+  int64_t partition_offset = shapes[0].GetDims()[0];
+  int64_t partition_size = shapes[1].GetDims()[0];
```

Comment:
Consider passing these as parameters explicitly, instead of packing into a vector of shapes.

Reply:
will change